src/fetch/FetchExample.re

```diff
@@ -0,0 +1,93 @@
+type tDog = string;
+
+type tArrayDogs = array(tDog);
+
+type state = { dogs: tArrayDogs, isLoading: bool, error: bool };
+
+type action =
+  | DogFetch
+  | DogFetched(tArrayDogs)
+  | DogFailedToFetch;
+
+let component = ReasonReact.reducerComponent("FetchExample");
+
+module Decode = {
+  let dogs = json : tArrayDogs =>
+    Array.map(
+      dog => dog,
+      json |> Json.Decode.field("message", 
+        Json.Decode.array(Json.Decode.string)
+      )
+    );
+};
+
+let make = _children => {
+  ...component,
+  initialState: () => { dogs: [||], isLoading: false, error: false },
+  reducer: (action, state) => 
+    switch action {
+    | DogFetch =>
+        ReasonReact.Update({
+          ...state,
+          isLoading: true
+        })
+    | DogFetched(dogs) =>
+      ReasonReact.Update({
+        ...state,
+        isLoading: false,
+        dogs
+      })
+    | DogFailedToFetch => 
+      ReasonReact.Update({
+        ...state,
+        error: true
+      })
+    },
+  didMount: self => {
+    self.send(DogFetch);
+
+    Js.Promise.(
```

Comment:
You migth want to either move this into the `DogFetch` action in the `reducer`, where you can use `UpdateWithSideEffects` to trigger the fetch, or set `isLoading` in `didMount` instead of returning `NoUpdate`. Separating the actual fetching from setting the loading state is more error-prone.